### ISSUE TITLE
Add 'present via pympress' to Explorer's context menu on *.pdf files on Windows

### ISSUE THREAD

**Issue by ComFreek** [Cimbali/pympress#152]
**Feature request:** when right-clicking on a PDF file in Explorer, show an entry "present via pympress"*. This saves me from

- starting pympress' GUI and using its *open file* dialog, which is hard to navigate (lacking ability of pasting a path, non-Windows-native UX)

- starting pympress on CLI for which I first need to figure out its path

**Does this make sense for other OS, too?**

*) or "open with pympress", there is probably a recommendation in Microsoft's UX guidelines

**Comment by Cimbali** [comment]
Not sure if cluttering the right-button menu would be universally welcomed, but I see the issue nonetheless. There’s maybe a way to register pympress with the OS as a pdf-handling program, to be shown in the right click > “Open with” submenu.

**Comment by ComFreek** [comment]
> There’s maybe a way to register pympress with the OS as a pdf-handling program, to be shown in the right click > “Open with” submenu.

Yes, registering Pympress as the PDF opening program would work, but I would not want that. The majority of PDFs I open are for reading, not for presenting.

**Comment by Cimbali** [comment]
I think it’s a more appropriate way to get the same result, no? In my imagination this doesn’t make pympress the _default_ pdf reader, just a _possible_ pdf reader.

Maybe using the [assoc](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/assoc) command in the post-install process could achieve that, but I’m don’t have a lot of time to look into this right now.

**Comment by ComFreek** [comment]
> I think it’s a more appropriate way to get the same result, no? In my imagination this doesn’t make pympress the default pdf reader, just a possible pdf reader.

You are right, I agree now!

> Maybe using the assoc command in the post-install process could achieve that

`assoc .pdf` gives me `.pdf=MiKTeX.pdf.2.9`, which is not even a program, so I doubt it is the best interface for modifying the open-with list. Additionally, it only works within cmd.exe, not even within PowerShell, so seems brittle to rely upon.

It seems one can modify the open-with list inside Windows' registry. It seems the way to go is to add a new verb handling program as [written here for Wix installers](https://stackoverflow.com/a/139168/603003). I am afraid that even after 10min of googling, I could not reliably find how this is done in terms of registry modifcations. Seems indeed to be require a bit of fiddling. Unless, of course, the `bdist_msi` thing you use for creating the MSI installers offers some options for this. I couldn't find any when googling for that, though.

**Comment by Cimbali** [comment]
Not easy this one. Seems like MSI installers can do that, but not out of the box with cx_Freeze. I tried to manually specify the Extension / MIME / Verb tables but to no avail so far. It seems doable with other installer builders, but maybe only with admin privileges.

**Comment by Cimbali** [comment]
![image](https://user-images.githubusercontent.com/6126377/114563348-a0eb2500-9c6f-11eb-8aad-bd52e8b41b74.png)

We’re getting closer, I finally found out how to do it… Just have to figure out how to adjust the text that is displayed now.

**Comment by Cimbali** [comment]
Done in f3a7fa5

**Comment by ComFreek** [comment]
Tested and works great so far!

I am only wondering why Pympress doesn't directly appear in the "open with" list, but rather I have to click "chose other app" (right context menu, lowest entry).
I have tried opening a PDF multiple times this way now (all successfully) to maybe make Windows move Pympress up a bit into this menu -- to no avail.
That TeXWorks is on this list (I never use it) would also support my conclusion now that this list isn't based on some sorting of application start frequencies at all.

Have you by chance found any way to include Pympress there?

![image](https://user-images.githubusercontent.com/1827709/114995113-c8273980-9e9d-11eb-855c-e409e4bc3e39.png)
